app/forms/incomeSources/cease/CeaseIncomeSourceEndDateFormProvider.scala

```diff
@@ -51,18 +51,18 @@ class CeaseIncomeSourceEndDateFormProvider extends Mappings {
 
     val minimumDate: Constraint[LocalDate] = incomeSourceType match {
       case UkProperty =>
-        val ukStartDate = user.incomeSources.properties.filter(_.isUkProperty).filter(!_.isCeased).flatMap(_.tradingStartDate)
-          .headOption.getOrElse(LocalDate.MIN)
+        val ukStartDate = user.incomeSources.properties.filter(_.isUkProperty).filter(!_.isCeased).map(_.getTradingStartDate).headOption.getOrElse(LocalDate.MIN)
```

Comment:
heya just wondering you know they were .filtering twice here, do you think it would be worth combining it into a single filter? Or is it different when it's filtered twice?

Reply:
Honestly i'm not sure, I can try to see if it'll break anything and get back to you on that

Reply:
yeah it looks like it breaks it because it needs to filter through to one type of income source first before it checks inside for what is ceased